DirectProgramming/DPC++FPGA/ReferenceDesigns/mvdr_beamforming/src/mvdr_beamforming.cpp

```diff
@@ -305,7 +305,11 @@ int main(int argc, char *argv[]) {
     // By waiting on this kernel to finish, we are assuring that the runtime
     // has already programmed the FPGA with the image for this kernel. This
     // makes calling SetupPAC below safe (for the real IO pipes).
-    SinThetaProducer::Start(q, kNumSteer).wait();
+    event steer_dma_event, steer_kernel_event;
+    std::tie(steer_dma_event, steer_kernel_event) =
+      SinThetaProducer::Start(q, kNumSteer);
+    steer_dma_event.wait();
+    steer_kernel_event.wait();
```

Comment:
Just for my own understanding - the kernel event already depends on the dma event, so it would be equally correct to just wait on the kernel event here, right?  I like this code anyway for clarity, just want to make sure I understand.

Reply:
This understanding is correct. I just wanted to be explicit.